lib/plugins/slack/summary.js

```diff
@@ -1,75 +1,74 @@
-        metric: base.browsertime ? base.browsertime.summary.pageTimings.domContentLoadedTime.median : undefined
-      },
-      speedIndex: {
-        name: 'Speed Index',
-        metric: base.browsertime ? base.browsertime.summary.visualMetrics ? base.browsertime.summary.visualMetrics.SpeedIndex.median : undefined : undefined
-      },
-      firstVisualChange: {
-        name: 'First Visual Change',
-        metric: base.browsertime ? base.browsertime.summary.visualMetrics ? base.browsertime.summary.visualMetrics.FirstVisualChange.median : undefined : undefined
-      },
-      visualComplete85: {
-        name: 'Visual Complete 85%',
-        metric: base.browsertime ? base.browsertime.summary.visualMetrics ? base.browsertime.summary.visualMetrics.VisualComplete85.median : undefined : undefined
-      },
-      lastVisualChange: {
-        name: 'Last Visual Change',
-        metric: base.browsertime ? base.browsertime.summary.visualMetrics ? base.browsertime.summary.visualMetrics.LastVisualChange.median : undefined : undefined
-      },
-      fullyLoaded: {
-        name: 'Fully Loaded',
-        metric: base.browsertime ? base.browsertime.summary.fullyLoaded ? base.browsertime.summary.fullyLoaded.median : undefined : undefined
-      },
-      coachScore: {
-        name: 'Coach score',
-        metric: base.coach ? base.coach.summary.performance.score.median : undefined
-      },
-      transferSize: {
-        name: 'Page transfer weight',
-        metric: base.pagexray ? h.size.format(base.pagexray.summary.transferSize.median) : undefined
-      }
+module.exports = function(dataCollection, options, errors) {
+  const base = dataCollection.summaryPages.index || {};
+  const metrics = {
+    firstPaint: {
+      name: 'First paint',
+      metric: get(base.browsertime, 'summary.firstPaint.median')
+    },
+    domContentLoadedTime: {
+      name: 'domContentLoadedTime',
+      metric: get(base.browsertime, 'summary.pageTimings.domContentLoadedTime.median')
+    },
+    speedIndex: {
+      name: 'Speed Index',
+      metric: get(base.browsertime, 'summary.visualMetrics.SpeedIndex.median')
+    },
+    firstVisualChange: {
+      name: 'First Visual Change',
+      metric: get(base.browsertime, 'summary.visualMetrics.FirstVisualChange.median')
+    },
+    visualComplete85: {
+      name: 'Visual Complete 85%',
+      metric: get(base.browsertime, 'summary.visualMetrics.VisualComplete85.median')
+    },
+    lastVisualChange: {
+      name: 'Last Visual Change',
+      metric: get(base.browsertime, 'summary.visualMetrics.LastVisualChange.median')
+    },
+    fullyLoaded: {
+      name: 'Fully Loaded',
+      metric: get(base.browsertime, 'summary.fullyLoaded.median')
+    },
+    coachScore: {
+      name: 'Coach score',
+      metric: get(base.coach, 'summary.performance.score.median')
+    },
+    transferSize: {
+      name: 'Page transfer weight',
+      metric: h.size.format(get(base.pagexray, 'summary.transferSize.median'))
```

Comment:
This is a slight change that might not be desirable, formatting a missing transfer size as N/A.